ArcadiaParties-SPA/src/app/department/department-selector/department-selector.component.html

```diff
@@ -0,0 +1,10 @@
+<form class="department-form">
+    <mat-form-field>
+        <mat-label>Department</mat-label>
+        <mat-select [(value)]="selectedDepartment" (selectionChange)="onChange()">
+            <mat-option *ngFor="let departament of ( departments$ | async)" [value]="departament">
```

Comment:
You can simplify it all and use department id as selector value. Probably, you won't need the whole department object in the application, because all API requests are made using department id and department name is used only in selector component.

Reply:
I've changed approach